backend/mora/common.py

```diff
@@ -50,6 +51,77 @@ def get(self, objid):
         pass
 
 
+class RequestType(enum.Enum):
+    CREATE = 0
+    EDIT = 1
+
+
+class Request(abc.ABC):
+
+    __slots__ = 'request', 'request_type', 'payload', 'uuid'
+
+    def __init__(self, request, request_type: RequestType):
+        """
+        Initialize a request, and perform all required validation.
+        :param request: A dict containing a request
+        :param request_type: A RequestType, either CREATE or EDIT
+        """
+        self.request_type = request_type
+        self.request = request
+        self.payload = None
+        self.uuid = None
+
+        if request_type == RequestType.CREATE:
+            self.create(request)
```

Comment:
Wouldn't it be better if the caller invoked these methods? I.e.:

```python
x = WhateverRequest(…).create(req)
```

Reply:
(Also, you're not calling super().__init__() in your method…)

Reply:
We could let the caller invoke the `create` methods, but I don't see how it would be better. Just different.
And I'll add a call to `__init__()`.